application/src/main/java/org/openhds/mobile/activity/NavigateActivity.java

```diff
@@ -322,12 +340,41 @@ private void hierarchySetup() {
             showDetailFragment();
             detailToggleFragment.setButtonHighlighted(true);
         }
+        deleteSubFormPaths();
+        populateFormsForPath();
 
         if(null != getCurrentVisit()){
             visitFragment.setButtonEnabled(true);
         }
     }
 
+    private void deleteSubFormPaths() {
```

Comment:
Naming here is confusing. It seems from reading the contents, this could actually be 'disassociateSentForms'?